src/daggy.js

```diff
@@ -131,6 +131,26 @@ const arrToString = (arr) => {
   return str + ')'
 }
 
+const makeConstructor= (fields, proto) => {
+  switch (fields.length) {
+    case 1: return function(a) { return makeValue(fields, proto, arguments) }
```

Comment:
Can you change this arguments in this function to `[a], [a, b] ...`, run benchmarks and post results? 
This should restore speed of `cata` and `toString`, also it wouldn't decrease creation time that much.